en/docs/install-and-setup/setup/api-controller/building-jenkins-ci-cd-pipeline-apictl.md

```diff
@@ -0,0 +1,328 @@
+
+    ```
+    $apictl init PetstoreAPI --oas https://raw.githubusercontent.com/OAI/OpenAPI-Specification/main/examples/v2.0/yaml/petstore.yaml
+
+    ```
+
+2. Generate deployment resource using the command given below.
+
+    ```
+    $apictl gen deployment-dir --source /Path/To/PetstoreAPI --destination <deployment_repo_path>/poc-cicd-deployment-repo
+    ```
+
+    You will see the resources created with the following structure.
+
+    ```
+    ├── DeploymentArtifacts_SwaggerPetstore-1.0.0
+    │   ├── api_meta.yaml
+    │   ├── certificates
+    │   └── params.yaml
+
+    ```
+
+3. Open the `params.yaml` file and update the content with the code given below.
+
+    ```
+    environments:
+      - name: dev
+        configs:
+          endpoints:
+              production:
+                  url: 'https://petstore.swagger.io/v2/'
+          deploymentEnvironments:
+              - displayOnDevportal: true
+                deploymentEnvironment: Default
+                deploymentVhost : localhost
+    ```
+    The environment name is set as `dev`. This configuration will be used for the `dev` environment.
+
+4. Commit this to the **dev** branch in the deployment repository.
+5. Copy the **PetstoreAPI** folder to the source repository git location. 
+6. Create a file named **meta.yaml** inside the **PetstoreAPI** folder and add a version similar to following.
+
+    ```
+    version: 1.0.0
+    ```
+    This version is used to deploy the bundle created using the **PetstoreAPI** API in the Artifactory repository.
+
+7. Following is the content for the PetstoreAPI. Commit all the files to the git repository
+
+    ```
+    ├── PetstoreAPI
+    │   ├── Definitions
+    │   │   └── swagger.yaml
+    │   ├── README.md
+    │   ├── api.yaml
+    │   ├── api_meta.yaml
+    │   ├── deployment_environments.yaml
+    │   └── meta.yaml
+
+    ```
+
+    When you commit to the source repository, the `CICD_ARTIFACT_UPLOAD` will get triggered first and upload the built component to the Artifactory. This is displayed in the console log of the Jenkins job.
+
+      <a href="{{base_path}}/assets/img/learn/api-controller/console-log-jenkins.png"><img src="{{base_path}}/assets/img/learn/api-controller/console-log-jenkins.png" alt="" width="70%"></a>
+
+    Once the bundle is uploaded to the Artifactory repository, it will trigger the deployment Jenkins job. The bundle getting deployed in API-M is displayed in the logs.
+
+      <a href="{{base_path}}/assets/img/learn/api-controller/apim-logs.png"><img src="{{base_path}}/assets/img/learn/api-controller/apim-logs.png" alt="" width="70%"></a>
+
+    Log in to the API Manager instance and you should see the API is created.
+
+      <a href="{{base_path}}/assets/img/learn/api-controller/created-api.png"><img src="{{base_path}}/assets/img/learn/api-controller/created-api.png" alt="" width="50%"></a>
```

Comment:
In this screenshot I can see it shows "2 APIs", but actually one is there. Any idea?

Reply:
No idea why it is not showing. Let me test it once again and update.